uswds/uswds-site
# USWDS - Header: Use recommended markup for site header and navigation

## Description

**Use recommended header and nav markup.** Header and navigation uses USWDS recommended extended header markup. The previous header/menu was having issues with talk back on android. This uses the latest recommended markup for header. Closes #1728.

## Additional information

### Major changes

- Use classes and markup from [USWDS extended header](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.app.cloud.gov/preview/uswds/uswds/develop/iframe.html?args=&id=components-header--extended&viewMode=story) markup ⋆
- Site header classes use BEM-style classes
- Removed unused or duplicate styles `uswds-custom.css` - `75KB → 74KB`
- Variables have been moved to custom project settings [04a08be50d3facb9565bfc26e44c160265320e6b]
- Mixins have their own partial now [e8cc4fb88c1f1a4ba88d2c7e8f51832a4cefd8d7]
- Site header styles moved to their own partial [fa1769223863bfe8f54eda2648d9636d3b98e9c9]
- Site search moved to an include [fb99215065df7f1a33794b6f54f48274c14d8a64]
- New dependencies
  - Added concurrently for more reliable parallel tasks [fd7c27ca163767618e79c7c14e583ce58f925cd1]
  - Ran `bundle update` [fd7c27ca163767618e79c7c14e583ce58f925cd1]

⋆ Not exact, because mobile menu needs accordion and desktop needs to be sticky.

### Screenshots 

#### Mobile

**Before**

![designsystem digital gov_](https://user-images.githubusercontent.com/3385219/182415103-25c9f832-4d78-4bb8-bd32-eec67190cddd.png)

**After**

![localhost_4000_](https://user-images.githubusercontent.com/3385219/182415124-92f521a9-42fc-4d6a-bd77-889265d6325f.png)

**Before**

![designsystem digital gov_ (1)](https://user-images.githubusercontent.com/3385219/182415225-d892bae3-769b-46c3-b34f-24cf5b1ac0be.png)

**After**

![localhost_4000_ (1)](https://user-images.githubusercontent.com/3385219/182415241-0845434c-5b4e-45a3-a7ba-039c68d1092d.png)

#### Desktop

**Before**

![image](https://user-images.githubusercontent.com/3385219/182415434-5be1e406-c715-4302-a9fe-fe28bc29f717.png)

**After**

![image](https://user-images.githubusercontent.com/3385219/182415528-dc332a21-cee0-4c7e-b5d6-534d44054012.png)

**Before**

![image](https://user-images.githubusercontent.com/3385219/182415716-c2e1b5e2-29b8-49a5-88ad-dba24df17394.png)

**After**

![image](https://user-images.githubusercontent.com/3385219/182415764-22e18747-83f0-459d-ae7b-093b3d22c5be.png)



### New dependencies

| Name         | Old | New   |
| :------------ | --- | ----- |
| concurrently | ―   | 7.3.0 |

## How to test

- [ ] Mobile menu should visually match production
- [ ] Desktop menu should visually match production
- [ ] Next report styles should be unaffected and look the same as production
- [ ] SASS should compile successfully
- [ ] Search should continue to function

---

Before you hit Submit, make sure you’ve done whichever of these applies to you:
